addons/account/models/account_move.py

```diff
@@ -220,6 +220,26 @@ def _sequence_year_range_monthly_regex(self):
         related="statement_line_id.statement_id"
     )
 
+    # === Adjusting Entries fields === #
+    adjusting_entries_origin_move_id = fields.Many2one(
+        comodel_name='account.move',
+        index='btree_not_null',
```

Comment:
This approach is inspired by a previous issue about improving deletion performance in large tables (here, `account.move`) containing many2one fields. and since this index is on a sparsely populated field, it's not costly :) 
https://github.com/odoo/odoo/pull/143852